addons/website_blog/static/src/js/contentshare.js

```diff
@@ -1,109 +1,143 @@
-                var selectedText = this.getSelection('string').substring(0, option.maxLength - baseLength - 23);
-
-                var text = window.btoa(encodeURIComponent(sprintf(tweet, selectedText, window.location.href)));
-                $popover_content.append(sprintf(
-                    "<a onclick=\"window.open('%s' + atob('%s'), '_%s','location=yes,height=570,width=520,scrollbars=yes,status=yes')\"><i class=\"ml4 mr4 fa fa-twitter fa-lg\"/></a>",
-                    option.shareLink, text, option.target));
-            }
-            return $popover_content;
-        },
-        commentEdition: function () {
-            $(".o_portal_chatter_composer_form textarea").val('"' + selected_text + '" ').focus();
-            const commentsEl = $('#o_wblog_post_comments')[0];
-            if (commentsEl) {
-                scrollTo(commentsEl).then(() => {
-                    window.location.hash = 'blog_post_comment_quote';
-                });
-            }
-        },
-        getSelection: function (share) {
-            if (window.getSelection) {
-                var selection = window.getSelection();
-                if (!selection || selection.rangeCount === 0) {
-                    return "";
-                }
-                if (share === 'string') {
-                    return String(selection.getRangeAt(0)).replace(/\s{2,}/g, ' ');
-                } else {
-                    return selection.getRangeAt(0);
-                }
-            } else if (document.selection) {
-                if (share === 'string') {
-                    return document.selection.createRange().text.replace(/\s{2,}/g, ' ');
-                } else {
-                    return document.selection.createRange();
-                }
-            }
-        },
-        popOver: function () {
-            this.destroy();
-            if (this.getSelection('string').length < option.minLength) {
-                return;
+        }
+    }
+
+    function getSelection(type) {
+        const selection = window.getSelection();
+        if (!selection || selection.rangeCount === 0) {
+            return "";
+        }
+        if (type === "string") {
+            return String(selection.getRangeAt(0)).replace(/\s{2,}/g, " ");
+        } else {
+            return selection.getRangeAt(0);
+        }
+    }
+
+    function popOver() {
+        destroy();
+        if (getSelection("string").length < option.minLength) {
+            return;
+        }
+        const data = getContent();
+        const range = getSelection();
+
+        const newNode = document.createElement("span");
+        range.insertNode(newNode);
+        newNode.className = option.className;
+
+        const popover = Popover.getOrCreateInstance(newNode, {
+            trigger: "manual",
+            placement: option.placement,
+            html: true,
+            content: () => data,
+        });
```

Comment:
Hello @odoo/rd-security, could we get an override here?
That popover is created within a nested function, and its content is generated in another nested function (`getContent()`) where each part of the link (the only thing that could use data from outside the function) is either escaped or uses `encodeURIComponent`.
See:
```js
const text = window.btoa(
    encodeURIComponent(sprintf(tweet, selectedTextShort, window.location.href))
)
```
```js
window.open(
    escape(option.shareLink) + decodedText,
    `_${escape(option.target)}`,
    "location=yes,height=570,width=520,scrollbars=yes,status=yes"
)
```